3-0-jpa-and-hibernate/3-1-3-author-book/src/test/java/com/bobocode/AuthorBookMappingTest.java

```diff
@@ -0,0 +1,308 @@
+package com.bobocode;
+
+import com.bobocode.model.Author;
+import com.bobocode.model.Book;
+import com.bobocode.util.EntityManagerUtil;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.hibernate.Session;
+import org.junit.jupiter.api.*;
+
+import javax.persistence.*;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class AuthorBookMappingTest {
+    private static EntityManagerUtil emUtil;
+    private static EntityManagerFactory entityManagerFactory;
+
+    @BeforeAll
+    static void setup() {
+        entityManagerFactory = Persistence.createEntityManagerFactory("BookAuthors");
+        emUtil = new EntityManagerUtil(entityManagerFactory);
+    }
+
+    @AfterAll
+    static void destroy() {
+        entityManagerFactory.close();
+    }
+
+    @Test
+    @Order(1)
+    @DisplayName("Save a book only")
+    void saveBookOnly() {
+        Book book = createRandomBook();
+
+        emUtil.performWithinTx(entityManager -> entityManager.persist(book));
+
+        assertThat(book.getId()).isNotNull();
+    }
+
+    @Test
+    @Order(2)
+    @DisplayName("Save book without a name")
+    void saveBookWithoutName() {
+        Book book = createRandomBook();
+        book.setName(null);
+
+        assertThatExceptionOfType(PersistenceException.class).isThrownBy(() -> emUtil.performWithinTx(entityManager ->
+                entityManager.persist(book)));
+    }
+
+    @Test
+    @Order(3)
+    @DisplayName("Save a Book with duplicated ISBN")
```

Comment:
@mathodcoast please use the same pattern for these display names. E.g. 
`save book throws expectation when ISBN is duplicate`
in other words
`method invocation does something when something`